In [1]:
import pandas as pd
import numpy as np

In [3]:
df = pd.read_csv("milknew.csv")

In [5]:
df

pH  Temprature  Taste  Odor  Fat   Turbidity  Colour   Grade
0     6.6          35      1     0     1          0     254    high
1     6.6          36      0     1     0          1     253    high
2     8.5          70      1     1     1          1     246     low
3     9.5          34      1     1     0          1     255     low
4     6.6          37      0     0     0          0     255  medium
...   ...         ...    ...   ...   ...        ...     ...     ...
1054  6.7          45      1     1     0          0     247  medium
1055  6.7          38      1     0     1          0     255    high
1056  3.0          40      1     1     1          1     255     low
1057  6.8          43      1     0     1          0     250    high
1058  8.6          55      0     1     1          1     255     low

[1059 rows x 8 columns]

# Describe each categorization to find differences and find important values

In [6]:
df.describe()

pH   Temprature        Taste         Odor         Fat   \
count  1059.000000  1059.000000  1059.000000  1059.000000  1059.000000   
mean      6.630123    44.226629     0.546742     0.432483     0.671388   
std       1.399679    10.098364     0.498046     0.495655     0.469930   
min       3.000000    34.000000     0.000000     0.000000     0.000000   
25%       6.500000    38.000000     0.000000     0.000000     0.000000   
50%       6.700000    41.000000     1.000000     0.000000     1.000000   
75%       6.800000    45.000000     1.000000     1.000000     1.000000   
max       9.500000    90.000000     1.000000     1.000000     1.000000   

         Turbidity       Colour  
count  1059.000000  1059.000000  
mean      0.491029   251.840415  
std       0.500156     4.307424  
min       0.000000   240.000000  
25%       0.000000   250.000000  
50%       0.000000   255.000000  
75%       1.000000   255.000000  
max       1.000000   255.000000

In [8]:
df[df['Grade'] == 'high'].describe()

pH  Temprature       Taste        Odor        Fat    Turbidity  \
count  256.000000  256.000000  256.000000  256.000000  256.000000  256.000000   
mean     6.692578   40.648438    0.664062    0.750000    0.996094    0.632812   
std      0.108752    3.739749    0.473242    0.433861    0.062500    0.482982   
min      6.500000   35.000000    0.000000    0.000000    0.000000    0.000000   
25%      6.600000   37.000000    0.000000    0.750000    1.000000    0.000000   
50%      6.700000   40.000000    1.000000    1.000000    1.000000    1.000000   
75%      6.800000   45.000000    1.000000    1.000000    1.000000    1.000000   
max      6.800000   45.000000    1.000000    1.000000    1.000000    1.000000   

           Colour  
count  256.000000  
mean   252.539062  
std      3.711989  
min    245.000000  
25%    250.000000  
50%    255.000000  
75%    255.000000  
max    255.000000

In [9]:
df[df['Grade'] == 'medium'].describe()

pH  Temprature       Taste        Odor        Fat    Turbidity  \
count  374.000000  374.000000  374.000000  374.000000  374.000000  374.000000   
mean     6.635027   39.721925    0.414439    0.163102    0.385027    0.125668   
std      0.130899    3.613751    0.493285    0.369953    0.487253    0.331919   
min      6.400000   34.000000    0.000000    0.000000    0.000000    0.000000   
25%      6.500000   37.000000    0.000000    0.000000    0.000000    0.000000   
50%      6.600000   38.000000    0.000000    0.000000    0.000000    0.000000   
75%      6.800000   45.000000    1.000000    0.000000    1.000000    0.000000   
max      6.800000   45.000000    1.000000    1.000000    1.000000    1.000000   

           Colour  
count  374.000000  
mean   250.336898  
std      5.262430  
min    240.000000  
25%    245.000000  
50%    255.000000  
75%    255.000000  
max    255.000000

In [17]:
df[df['Grade'] == 'low'].describe()

pH  Temprature       Taste        Odor        Fat    Turbidity  \
count  429.000000  429.000000  429.000000  429.000000  429.000000  429.000000   
mean     6.588578   50.289044    0.592075    0.477855    0.727273    0.724942   
std      2.194712   13.043794    0.492023    0.500093    0.445882    0.447065   
min      3.000000   34.000000    0.000000    0.000000    0.000000    0.000000   
25%      4.700000   40.000000    0.000000    0.000000    0.000000    0.000000   
50%      6.800000   45.000000    1.000000    0.000000    1.000000    1.000000   
75%      8.600000   55.000000    1.000000    1.000000    1.000000    1.000000   
max      9.500000   90.000000    1.000000    1.000000    1.000000    1.000000   

           Colour  
count  429.000000  
mean   252.734266  
std      3.233046  
min    245.000000  
25%    250.000000  
50%    255.000000  
75%    255.000000  
max    255.000000

In [77]:
df.nunique()

pH            16
Temprature    17
Taste          2
Odor           2
Fat            2
Turbidity      2
Colour         9
Grade          3
dtype: int64

### Based from the results of the means we can have the following conclusion:
- A milk with a grade of *high* almost certainly has fat
- A milk with a grade of *medium* may or may not have fat
- A milk with a grade of *low* may have fat but is less likely to have one compared to a *high* grade

- It appears that taste does not matter in any of the grades as it appears to be almost perfectly random with high standard deviation on each of them.

- With pH, it appears to matter with *low* grade having a pH level of around 6.59, *medium* grade around 6.64, and *high* grade around 6.69 with little deviation of approximately 0.1 except for *low* grade with 2.19 standard deviation. 

- Temprature also seems to matter between *high* and *medium* grade milk with low standard deviation compared to the *low* grade with very high standard deviation, it also appears that both *high* grade and *medium* grade milk almost have the same temprature compared to the *low* grade milk.

- For odor, a *high* grade milk will likely have odor, a *medium* grade milk will unlikely have odor, and for a *low* grade milk, odor does not matter. 

- For turbidity, a *medium* grade milk will likely have none, whilst a *low* and *high* grade milk will either have one or not, where-in a *low* grade milk is more likely to have turbidity.

- As for colour, both *low* and *high* grade milk have almost the same colour, whilst the *medium* grade is different from the two of them.

### With these following insights, we can proceed to give the following importance:
1. Fat Content
2. pH level
3. Temprature
4. Turbidity
5. Odor
6. Colour
7. Taste


### A simple 